src/sqlfluff/rules/L010.py

```diff
@@ -100,6 +103,12 @@ def _eval(self, context: RuleContext) -> LintResult:
         if ignore_words_list and context.segment.raw.lower() in ignore_words_list:
             return LintResult(memory=context.memory)
 
+        # Skip if matches ignore regex
+        if self.ignore_words_regex and regex.search(
```

Comment:
I notice we're using `search()` rather than `match()`.
* Should we consider using `match()`? (Less risk of accidental matches causing confusing behavior.)
* If not, should we document that the patterns should probably either start with `^`, end with `$`, or both?

Reply:
Look in comments above @barrywhart. Search seemed easier on users than match.

Reply:
I'm okay with ‘search()`, but I do think we should document it. It could be simple, just an example in the rule docs page or a commented-out example (starting with `^`) in the `.py` file that documents what the option means.

Reply:
I like the idea of just clarifying it in the docs. I added small description.